src/commands/createWebApp/createWebApp.ts

```diff
@@ -3,22 +3,92 @@
+import { SetPostPromptDefaultsStep } from "./SetPostPromptDefaultsStep";
+import { setPrePromptDefaults } from "./setPrePromptDefaults";
+import { getCreatedWebAppMessage, showCreatedWebAppMessage } from "./showCreatedWebAppMessage";
+import { WebAppStackStep } from "./stacks/WebAppStackStep";
+import { WebAppCreateStep } from "./WebAppCreateStep";
 
-export async function createWebApp(context: IActionContext & Partial<ICreateChildImplContext>, node?: AzExtParentTreeItem | undefined, suppressCreatedWebAppMessage: boolean = false): Promise<SiteTreeItem> {
+export async function createWebApp(context: IActionContext & Partial<ICreateChildImplContext>, node?: SubscriptionTreeItemBase | undefined, suppressCreatedWebAppMessage: boolean = false): Promise<SiteTreeItem> {
     if (!node) {
-        node = <AzExtParentTreeItem>await ext.rgApi.tree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
+        node = <SubscriptionTreeItemBase>await ext.rgApi.tree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
+    }
+
+    const wizardContext: IWebAppWizardContext = Object.assign(context, node.subscription, {
+        newSiteKind: AppKind.app,
+        resourceGroupDeferLocationStep: true,
+        ...(await createActivityContext())
+    });
+
+    await setPrePromptDefaults(wizardContext);
+
+    const promptSteps: AzureWizardPromptStep<IWebAppWizardContext>[] = [];
+    const executeSteps: AzureWizardExecuteStep<IWebAppWizardContext>[] = [];
+    const siteStep: SiteNameStep = new SiteNameStep();
+    promptSteps.push(siteStep);
+
+    if (context.advancedCreation) {
+        promptSteps.push(new ResourceGroupListStep());
+        promptSteps.push(new WebAppStackStep());
+        CustomLocationListStep.addStep(wizardContext, promptSteps);
+        promptSteps.push(new AppServicePlanListStep());
+        promptSteps.push(new AppInsightsListStep());
+    } else {
+        promptSteps.push(new WebAppStackStep());
+        promptSteps.push(new AppServicePlanSkuStep());
+        LocationListStep.addStep(wizardContext, promptSteps);
+        executeSteps.push(new ResourceGroupCreateStep());
+        executeSteps.push(new AppServicePlanCreateStep());
+        executeSteps.push(new AppInsightsCreateStep());
+        executeSteps.push(new SetPostPromptDefaultsStep(siteStep));
+    }
+
+    executeSteps.push(new VerifyProvidersStep([webProvider, 'Microsoft.Insights']));
+    executeSteps.push(new WebAppCreateStep());
+
+    if (wizardContext.newSiteOS !== undefined) {
+        await setLocationsTask(wizardContext);
+    }
+
+    const title: string = localize('createApp', 'Create new web app');
+    const wizard: AzureWizard<IWebAppWizardContext> = new AzureWizard(wizardContext, { promptSteps, executeSteps, title });
+
+    await wizard.prompt();
+
+    const newSiteName = nonNullProp(wizardContext, 'newSiteName');
+
+    wizardContext.activityTitle = localize('createWebApp', 'Create web app "{0}"...', newSiteName);
```

Comment:
```suggestion
    wizardContext.activityTitle = localize('createWebApp', 'Creating web app "{0}"...', newSiteName);
```

Reply:
So I've actually been using the "Create web app" or "Delete web app" instead of "Deleting web app" and "Creating web app" for the activityTitle because I'm keeping the title the same for the entire life of the activity.

The activity title is "Create web app" regardless of if the activity is running, failed, or succeeded. But the progress message is often "Creating web app..."

Like this:
![image](https://user-images.githubusercontent.com/12476526/164293360-0d12cbce-7f26-41a5-b8ba-9199211f02b5.png)

Reply:
That's a good idea, I retract my suggestion. :smile: